src/gutter-container-component.coffee

```diff
@@ -0,0 +1,84 @@
+CustomGutterComponent = require './custom-gutter-component'
+LineNumberGutterComponent = require './line-number-gutter-component'
+
+# The GutterContainerComponent manages the GutterComponents of a particular
+# TextEditorComponent.
+
+module.exports =
+class GutterContainerComponent
+
+  constructor: ({@onLineNumberGutterMouseDown, @editor}) ->
+    @gutterComponents = []
+    @gutterComponentsByGutterName = {}
+    @lineNumberGutterComponent = null
+
+    @domNode = document.createElement('div')
+    @domNode.classList.add('gutter-container')
+    @domNode.style.display = 'flex';
+
+  getDomNode: ->
+    @domNode
+
+  getLineNumberGutterComponent: ->
+    @lineNumberGutterComponent
+
+  updateSync: (state) ->
+    # The GutterContainerComponent expects the gutters to be sorted in the order
+    # they should appear.
+    newState = state.gutters.sortedDescriptions
+
+    newGutterComponents = []
+    newGutterComponentsByGutterName = {}
+    for {gutter, visible} in newState
```

Comment:
Move decoration to be within this object.
Update the gutterComponent with only the state it needs. It can handle its own visibility.